File: back_end/internal/usecase/dto/task_dto.go
package dto

type TaskInputDTO struct {
	Title       string `json:"title"`
	Description string `json:"description"`
	Priority    bool   `json:"priority"`
	ListID      string `json:"list_id"`
	TimeSelect  string `json:"time_select"`
}

type TaskOutputDTO struct {
	ID          string `json:"id"`
	Title       string `json:"title"`
	Description string `json:"description"`
	Status      bool   `json:"status"`
	Priority    bool   `json:"priority"`
	ListID      string `json:"list_id"`
	IsDeleted   bool   `json:"is_deleted"`
	TimeSelect  string `json:"time_select"`
}

type TaskFindTitleInputDTO struct {
	Title string `json:"title"`
}

type TaskOutputMessageDTO struct {
	Message string `json:"mensage"`
}

type TaskInputSoftDeleteDTO struct {
	ID string `json:"id"`
}
type TaskInputCompletedDTO struct {
	ID     string `json:"id"`
	Status bool   `json:"status"`
}

type TaskEditInputDTO struct {
	ID          string `json:"id"`
	Title       string `json:"title"`
	Description string `json:"description"`
	Priority    bool   `json:"priority"`
	ListID      string `json:"list_id"`
	TimeSelect  string `json:"time_select"`
}

type TaskEditOutputDTO struct {
	ID          string `json:"id"`
	Title       string `json:"title"`
	Description string `json:"description"`
	Priority    bool   `json:"priority"`
	ListID      string `json:"list_id"`
	UpdatedAt   string `json:"update_at"`
	TimeSelect  string `json:"time_select"`
}

File: front_end/services/dto/task_dto.ts
export interface InputCreateTaskDto {
    title: string;
    description?: string;
    priority?: boolean;
    list_id: string;
    time_select?: string;
}
export interface TaskIdInputDto {
    id: string;
}
export interface TaskTitleInputDto {
    title: string;
}
export interface TaskStatusInputDto {
    id: string;
    status: boolean;
}
export interface TaskCompletedInputDto {
    id: string;
    status: boolean;
}

export interface OutputTaskDto {
    id: string;
    title: string;
    description: string;
    status: boolean;
    priority: boolean;
    list_id: string;
    time_select:string;
    is_deleted: boolean;
}

